### PR TITLE
Add build script

### PR DESCRIPTION
**Pull request by r8d8**
Related to #429

### CODE REVIEW

**Comment by whilei** [comment]
Looks like this doesn't include the build steps for sputnik yet -- that on your horizon?

**Comment by whilei** [comment]
See https://github.com/r8d8/go-ethereum/pull/1 for a few minor edits

File: scripts/build_sputnikvm.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	echo "Building SputnikVM FFI..."
+    cargo build --release
+    cp $sputnikffi_path/c/ffi/target/release/libsputnikvm_ffi.a \
+        $sputnikffi_path/c/libsputnikvm.a
+
+	geth_binpath="$ep_gopath/go-ethereum/bin"
+	echo "Building geth to $geth_binpath/geth..."
+	mkdir -p "$geth_binpath"
+    if [ "$OS" == "Linux" ]; then
+        cd $GOPATH/src/github.com/ethereumproject/go-ethereum/cmd/geth
```
Comment (whilei) [inline scripts/build_sputnikvm.sh]: We can also get rid of these `cd` lines if we use `go build [...] ./cmd/geth`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+build: ## Build a local snapshot binary version
+	./scripts/build.sh ${LDFLAGS} ${BINARY}
+
+sputnikvm: ## Build a local snapshot binary version with sputnikvm
+	./scripts/build_sputnikvm.sh ${LDFLAGS} ${BINARY}
+
+clean: ## Remove a local snapshot binary version
+	if [ -d ${BINARY} ] ; then rm -rf ${BINARY} ; fi
+
+install: ## Install to $GOPATH/src
```
Comment (whilei) [inline Makefile]: should be `$GOPATH/bin/`

**Comment by whilei** [comment]
Couple of questions/ideas:

`make build` and `make install` are brothers; both can/could be used not only for geth, but for any `./cmd/` package. It seems like it would make sense to use `/...` as a default for both, but add an optional param/arg to choose one (or multiple?) in particular, eg.
```
make install geth
make build geth
make install geth,abigen # this might be overkill, and could be a follow-up nice-to-have feature
```

Since, at this point, sputnikvm is included in the distribution builds, it seems like it should be included with the default `make build geth`, but toggle-able, eg.
```
make install geth --sputnikvm=[0|false|1|true] # default 1/true
make build geth --sputnikvm=[0|false|1|true] # default 1/true
```

**Comment by whilei** [comment]
Also I think we should changes updating the README with revised build instructions

**Comment by whilei** [comment]
But I think wise to keep any changes to CI configs in separate PR

File: scripts/build_sputnikvm.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+				git pull %remote_name% master
+			fi
+		fi
+    	cd c\ffi
+	else
+    	git clone https://github.com/ethereumproject/sputnikvm-ffi.git
+    	cd sputnikvm-ffi\c\ffi
+	fi
+    cargo build --release
+    copy %GOPATH%\src\github.com\ethereumproject\sputnikvm-ffi\c\ffi\target\release\sputnikvm.lib \
```
Comment (whilei) [inline scripts/build_sputnikvm.sh]: just an idea to consider would be to refactor this to something like

Grab and build all dependencies for installing/building geth with SputnikVM, but don't build or install
```shell
make sputnikvm
```


```shell
make build geth 
make build geth SPUTNIKVM=1 # runs `make sputnikvm` to ensure/update svm dependencies, then builds geth
```

just as a way to maybe keep things DRY/modular (?)

File: scripts/build_sputnikvm.sh (hunk 1 above)
Comment (whilei) [reply scripts/build_sputnikvm.sh]: but see the idea about refactoring for this too

File: scripts/build_sputnikvm.sh (hunk 3 above)
Comment (r8d8) [reply scripts/build_sputnikvm.sh]: I was thinking to make docs and CI builds updates as separate PRs

**Comment by r8d8** [comment]
I was thinking to make `docs` and `CI builds` updates as separate PRs

**Comment by r8d8** [comment]
My idea was to provide `make build` command for dev usage, all binaries will be build into local <repo>/bin directory, not to mess them with system wide binaries

**Comment by whilei** [comment]
I did a little poking around about passing args to `make <target>` commands it seems it's not the best thing (erm... [you're breaking the basic pattern of how Make works](https://stackoverflow.com/questions/6273608/how-to-pass-argument-to-makefile-from-command-line#6273809))

**Comment by whilei** [comment]
Checkout https://github.com/r8d8/go-ethereum/pull/2 for a slightly revised UI

**Comment by whilei** [comment]
We considered building SputnikVM from dependencies instead of external repo... is that still a gameplan?

**Comment by r8d8** [comment]
@whilei, 
sorry I missing point
You asking, does this makefile supports compiling `geth` with  SputnikVM used  from external repo?

**Comment by whilei** [comment]
Oh, I mean currently `scripts/build_sputnikvm.sh` compiles sputnik sorces from `copy %GOPATH%\src\github.com\ethereumproject\sputnikvm-ffi\c\ffi\target\release\sputnikvm.lib`, which is in "external" repo, but we have existing copy in https://github.com/ethereumproject/go-ethereum/tree/master/vendor/github.com/ethereumproject/sputnikvm-ffi, so could also compile from there for dependency safety.

**Comment by whilei** [comment]
@r8d8 plz rebase/merge 🍰

**Pull request merged**